rust/lancedb/src/table.rs

```diff
@@ -1756,6 +1757,37 @@ impl TableInternal for NativeTable {
             .await
     }
 
+    async fn explain_plan(&self, query: &VectorQuery, verbose: bool) -> Result<String> {
```

Comment:
I feel like this replicates some (but not all) of the logic from `create_plan` above. Could we instead pull this out to some common function (`build_plan`?) and then both `create_plan` and `explain_plan` can be built off of those?

Reply:
Good point. Pulled out common functionality here: https://github.com/lancedb/lancedb/pull/1328/commits/6dc8ced985c1798edf3c5f927d80335ae561716a